File: cs/libldt3/model/saetze/Auftrag.cs
/*
 * Copyright 2016-2024  Christoph Brill <opensource@christophbrill.de>
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
using libldt3.attributes;
using libldt3.model;
using libldt3.model.enums;
using libldt3.model.objekte;
using libldt3.model.regel.kontext;

namespace libldt3
{
    namespace model
    {
        namespace saetze
        {
            /// <summary>
            /// Satzart: Auftrag "8215"
            /// </summary>
            [Datenpaket(Value = Satzart.Auftrag, Kontextregeln = new[] { typeof(K057) })]
            public class Auftrag : Satz, Kontext
            {
                [Feld(Value = "8122", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(Laenge = 23)]
                public Einsenderidentifikation Einsenderidentifikation;
                [Feld(Value = "8145", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(Laenge = 7)]
                public Patient Patient;
                [Feld(Value = "8169", Feldart = Feldart.bedingt_kann)]
                [Regelsatz(Laenge = 19)]
                public Koerperkenngroessen Koerperkenngroessen;
                [Feld(Value = "8150", Feldart = Feldart.bedingt_kann)]
                [Regelsatz(Laenge = 15)]
                public Schwangerschaft Schwangerschaft;
                [Feld(Value = "8140", Feldart = Feldart.bedingt_kann)]
                [Regelsatz(Laenge = 12)]
                public Mutterschaft Mutterschaft;
                [Feld(Value = "8153", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(Laenge = 14)]
                public TierSonstiges TierSonstiges;
                [Feld(Value = "8113", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(Laenge = 19)]
                public Auftragsinformation Auftragsinformation;
                [Feld(Value = "8127", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(Laenge = 18)]
                public IList<Veranlassungsgrund> Veranlassungsgrund;
                [Feld(Value = "8101", Feldart = Feldart.muss)]
                [Regelsatz(Laenge = 22)]
                public Abrechnungsinformationen Abrechnungsinformation;
                [Feld(Value = "8137", Feldart = Feldart.bedingt_kann)]
                [Regelsatz(Laenge = 8)]
                public IList<Material> Material;
                [Feld(Value = "8159", Feldart = Feldart.bedingt_muss)]
                [Regelsatz(Laenge = 24)]
                public IList<Untersuchungsanforderung> Untersuchungsanforderung;
                [Feld(Value = "8167", Feldart = Feldart.kann)]
                [Regelsatz(Laenge = 26)]
                public IList<Fliesstext> ZusaetzlicheInformationen;
                [Feld(Value = "8110", Feldart = Feldart.kann)]
                [Regelsatz(Laenge = 6)]
                public IList<Anhang> Anhang;

            }
        }
    }
}

File: java/src/main/java/libldt3/model/saetze/Auftrag.java
/*
 * Copyright 2016-2024  Christoph Brill &lt;opensource@christophbrill.de&gt;
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package libldt3.model.saetze;

import java.util.List;

import libldt3.annotations.Datenpaket;
import libldt3.annotations.Feld;
import libldt3.annotations.Feldart;
import libldt3.annotations.Regelsatz;
import libldt3.model.Kontext;
import libldt3.model.enums.Satzart;
import libldt3.model.objekte.Abrechnungsinformationen;
import libldt3.model.objekte.Anhang;
import libldt3.model.objekte.Auftragsinformation;
import libldt3.model.objekte.Einsenderidentifikation;
import libldt3.model.objekte.Fliesstext;
import libldt3.model.objekte.Koerperkenngroessen;
import libldt3.model.objekte.Material;
import libldt3.model.objekte.Mutterschaft;
import libldt3.model.objekte.Patient;
import libldt3.model.objekte.Schwangerschaft;
import libldt3.model.objekte.TierSonstiges;
import libldt3.model.objekte.Untersuchungsanforderung;
import libldt3.model.objekte.Veranlassungsgrund;
import libldt3.model.regel.kontext.K057;

/**
 * Satzart: Auftrag "8215"
 */
@Datenpaket(value = Satzart.Auftrag, kontextregeln = K057.class)
public class Auftrag implements Satz, Kontext {

    @Feld(value = "8122", feldart = Feldart.bedingt_muss)
    @Regelsatz(laenge = 23)
    public Einsenderidentifikation einsenderidentifikation;
    @Feld(value = "8145", feldart = Feldart.bedingt_muss)
    @Regelsatz(laenge = 7)
    public Patient patient;
    @Feld(value = "8169", feldart = Feldart.bedingt_kann)
    @Regelsatz(laenge = 19)
    public Koerperkenngroessen koerperkenngroessen;
    @Feld(value = "8150", feldart = Feldart.bedingt_kann)
    @Regelsatz(laenge = 15)
    public Schwangerschaft schwangerschaft;
    @Feld(value = "8140", feldart = Feldart.bedingt_kann)
    @Regelsatz(laenge = 12)
    public Mutterschaft mutterschaft;
    @Feld(value = "8153", feldart = Feldart.bedingt_muss)
    @Regelsatz(laenge = 14)
    public TierSonstiges tierSonstiges;
    @Feld(value = "8113", feldart = Feldart.bedingt_muss)
    @Regelsatz(laenge = 19)
    public Auftragsinformation auftragsinformation;
    @Feld(value = "8127", feldart = Feldart.bedingt_muss)
    @Regelsatz(laenge = 18)
    public List<Veranlassungsgrund> veranlassungsgrund;
    @Feld(value = "8101", feldart = Feldart.muss)
    @Regelsatz(laenge = 22)
    public Abrechnungsinformationen abrechnungsinformation;
    @Feld(value = "8137", feldart = Feldart.bedingt_kann)
    @Regelsatz(laenge = 8)
    public List<Material> material;
    @Feld(value = "8159", feldart = Feldart.bedingt_muss)
    @Regelsatz(laenge = 24)
    public List<Untersuchungsanforderung> untersuchungsanforderung;
    @Feld(value = "8167", feldart = Feldart.kann)
    @Regelsatz(laenge = 26)
    public List<Fliesstext> zusaetzlicheInformationen;
    @Feld(value = "8110", feldart = Feldart.kann)
    @Regelsatz(laenge = 6)
    public List<Anhang> anhang;

}
